thentos-core/src/Thentos/Action.hs

```diff
@@ -613,6 +615,16 @@ getServiceSessionMetadata :: ServiceSessionToken -> Action e s ServiceSessionMet
 getServiceSessionMetadata tok = (^. srvSessMetadata) <$> lookupServiceSession tok
 
 
+-- * send emails
+
+-- | Send an email. Only a privileged IP is allowed to use this endpoint.
+sendEmail :: SendEmailRequest -> Action e s ()
+sendEmail req = do
+    void hasPrivilegedIP
```

Comment:
I wonder if `hasPrivilegedIP` should be changed to make this `void` unnecessary?  (Can't be bothered to check myself, not necessarily part of this PR, just wondering...)